src/pykeen/stoppers/early_stopping.py

```diff
@@ -17,52 +15,43 @@
 from ..utils import fix_dataclass_init_docs
 
 __all__ = [
-    'smaller_than_any_buffer_element',
-    'larger_than_any_buffer_element',
+    'is_improvement',
     'EarlyStopper',
     'StopperCallback',
 ]
 
 logger = logging.getLogger(__name__)
 
+StopperCallback = Callable[[Stopper, Union[int, float]], None]
 
-def smaller_than_any_buffer_element(buffer: numpy.ndarray, result: float, delta: float = 0.) -> bool:
-    """Decide if a result is better than at least one buffer element, where smaller is better.
-
-    :param buffer:
-        The last results to compare against (excluding the current result).
-    :param result:
-        The current result.
-    :param delta:
-        The minimum improvement.
 
-    :return:
-        Whether the result is at least delta better than at least one value in the buffer.
+def is_improvement(
+    best_value: float,
+    current_value: float,
+    larger_is_better: bool,
+    absolute_delta: float = 0.0,
+) -> bool:
     """
-    worst_in_window = buffer.max()
-    baseline = worst_in_window - delta
-    return result < baseline
-
+    Decide whether the current value is an improvement over the best value.
 
-def larger_than_any_buffer_element(buffer: numpy.ndarray, result: float, delta: float = 0.) -> bool:
-    """Decide if a result is better than at least one buffer element, where larger is better.
-
-    :param buffer:
-        The last results to compare against (excluding the current result).
-    :param result:
-        The current result.
-    :param delta:
-        The minimum improvement.
+    :param best_value:
+        The best value so far.
+    :param current_value:
+        The current value.
+    :param larger_is_better:
+        Whether a larger value is better.
+    :param absolute_delta:
+        A minimum absolute improvement until it is considered as an improvement.
 
     :return:
-        Whether the result is at least delta better than at least one value in the buffer.
+        Whether the current value is better.
     """
-    worst_in_window = buffer.min()
-    baseline = worst_in_window + delta
-    return result > baseline
-
+    if larger_is_better:
+        best_value = -best_value
```

Comment:
I would prefer a different return here that is 

```python
return -current_value < (-best_value - absolute_delta)
```

Or simplified

```python
return current_value > best_value + absolute_delta
```

Reply:
1664e2d 

I also changed from using an `absolute_delta` to a relative one, which should be easier to set.